code/game/objects/effects/effect_system/effects_foam.dm

```diff
@@ -305,14 +307,70 @@
 	to_chat(user, span_warning("You hit [src] but bounce off it!"))
 	playsound(src.loc, 'sound/weapons/tap.ogg', 100, TRUE)
 
+/obj/structure/foamedmetal/attackby(obj/item/W, mob/user, params)
+	///A speed modifier for how fast the wall is build
+	var/platingmodifier = 1
+	if(HAS_TRAIT(user, TRAIT_QUICK_BUILD))
+		platingmodifier = 0.7
+		if(next_beep <= world.time)
+			next_beep = world.time + 10
+			playsound(src, 'sound/machines/clockcult/integration_cog_install.ogg', 50, TRUE)
+	add_fingerprint(user)
+
+	if(!istype(W, /obj/item/stack/sheet))
+		return ..()
+
+	var/obj/item/stack/sheet/sheet_for_plating = W
+	if(istype(sheet_for_plating, /obj/item/stack/sheet/iron))
+		if(sheet_for_plating.get_amount() < 2)
+			to_chat(user, span_warning("You need two sheets of iron to finish a wall!"))
+			return
+		to_chat(user, span_notice("You start adding plating to the foam structure..."))
+		if (do_after(user, 40*platingmodifier, target = src))
+			if(sheet_for_plating.get_amount() < 2)
+				return
+			sheet_for_plating.use(2)
```

Comment:
IIRC, `use()` already checks if the stack has at least the specified amount. So you could just do

```
if(!sheet_for_plating.use(2))
  return

Reply:
Done and thanks.